### PR TITLE
Improve collection scroll behavior

### LINKED ISSUE #53: Improve scrolling behavior when using pagination

### LINKED ISSUE #63: Reset collection pagination when filters change
## The problem

When the customer navigates through pages of a collection, and then he changes the filters, often he gets no results even though there are results.

## Possible solution

This happen because the filtered collection has less pages than before, and the page currently selected by the user does not exist in the filtered set. We can just reset the page when filters or search query change.

### PR DESCRIPTION
**Pull request by dhmacs**
This PR closes #53, #54 and fixes #63 

## QA

1. Visit the Vercel PR preview
2. Visit any collection that contains products
3. Verify that when you change the page of the product list, the top item is scrolled into view
4. Verify that when you change filters or search query, the top item is scrolled into view

### CODE REVIEW

**Comment by k0rvusk0r4x** [comment]
Where would we find the Vercel PR preview?

**Comment by cdcline** [comment]
CR 👍

**Comment by dhmacs** [comment]
@k0rvusk0r4x there's a comment posted by the vercel bot above https://github.com/iFixit/react-commerce/pull/64#issuecomment-894324671

**Comment by ardelato** [comment]
QA 👍 

## Steps Taken:
1. Clicked on the Vercel Preview Link in the GitHub comment above
2. I then confirmed the issues were fixed
3. Improve scrolling behavior when using pagination #53

   <details>
	<summary> Before </summary>

   https://user-images.githubusercontent.com/22064420/126721177-0c5ffca2-525e-48c8-a425-dbe5336bd471.mp4


   </details>

   <details>
	<summary>After </summary>

   https://user-images.githubusercontent.com/22064420/128561816-182b64d8-aa77-4359-96a4-0a9555272492.mp4

   </details>

4. Improve scrolling behavior of products list when selecting filter options #54

   <details>
	<summary> Before </summary>

   https://user-images.githubusercontent.com/22064420/126721336-228ee845-d531-431f-8a74-ed173b123460.mp4


   </details>

   <details>
	<summary>After </summary>


   https://user-images.githubusercontent.com/22064420/128562182-b5a9bc06-516d-49f1-96c7-2107d886115d.mp4



   </details>

5. Reset collection pagination when filters change #63

   <details>
	<summary> Before </summary>


   https://user-images.githubusercontent.com/22064420/128562275-3bce6285-d51d-44d4-8dba-309fbae88d8e.mp4



   </details>

   <details>
	<summary> After </summary>



   https://user-images.githubusercontent.com/22064420/128562312-75e492dd-05bc-4599-a4aa-1b11d6919fc9.mp4

   </details>

6. I also confirmed that pagination with the search filter still worked

   <details>
	<summary> Example </summary>


   https://user-images.githubusercontent.com/22064420/128562423-596dd6c2-a0f8-4925-b158-94879815ea38.mp4


   </details>
7. Lastly, I did a quick smoke test verifying general functionality remained the same for filtering and pagination.

File: features/collection/components/FilterableProductSection.tsx [hunk 1]
```diff
@@ -163,6 +170,21 @@ function useCollectionState(): CollectionState {
...
+function useScrollIntoViewEffect(deps: React.DependencyList) {
+   const ref = React.useRef<HTMLDivElement>(null);
+   React.useEffect(() => {
+      if (ref.current) {
+         if (ref.current.getBoundingClientRect().top < 0) {
+            ref.current.scrollIntoView({
+               behavior: 'smooth',
+            });
+         }
+      }
```
Comment (zdmitchell) [inline features/collection/components/FilterableProductSection.tsx:182]:
```suggestion
      if (ref.current && ref.current.getBoundingClientRect().top < 0) {
         ref.current.scrollIntoView({
            behavior: 'smooth',
         });
      }
```

**Comment by zdmitchell** [comment]
CR :guitar: but deploy_block :guitar: on comment.

**Pull request merged**